core/src/main/java/org/apache/calcite/util/TimestampWithTimeZoneString.java

```diff
@@ -41,21 +46,39 @@ public class TimestampWithTimeZoneString
   final TimestampString localDateTime;
   final TimeZone timeZone;
   final String v;
+  private final DateTimeUtils.PrecisionTime pt;
+
+  private static final ThreadLocal<@Nullable SimpleDateFormat> TIMESTAMP_FORMAT =
```

Comment:
We can confirm TimestampWithTimeZoneString##constructor would be call by multiple thread?  If not we may not need to use ThreadLocal.

Reply:
It not easy, also if it thread safety right now we have no guarantee that it remain same in future.
DateFormat is not thread safe (check java doc) and checker framework reports about erroneous usage without thread local mechanics. Thus i forced to use such a code, if you have different approach here - plz let me know. Thanks !